code/modules/reagents/reactions/distilling/distilling.dm

```diff
@@ -26,32 +26,19 @@
 	var/list/temp_range = list(T0C, T20C)
 	var/temp_shift = 0 // How much the temperature changes when the reaction occurs.
 
-/datum/chemical_reaction/distilling/can_happen(var/datum/reagents/holder)
-	//check that all the required reagents are present
-	if(!holder.has_all_reagents(required_reagents))
-		return 0
+/decl/chemical_reaction/distilling/can_happen(var/datum/reagents/holder)
+	if(!(istype(holder, /datum/reagents/distilling) && istype(holder.my_atom, /obj/machinery/portable_atmospherics/powered/reagent_distillery)))
```

Comment:
Purely a nitpick, but `(!(foo && bar))` always seems less readable than `(!foo || !bar)` to me.